src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -510,7 +510,9 @@ private void CheckRequestBeforeSend(HttpRequestMessage request)
                 // Wait for the send request to complete, getting back the response.
                 response = async ?
                     await base.SendAsync(request, cts.Token).ConfigureAwait(false) :
+#pragma warning disable // Validate platform compatibility
                     base.Send(request, cts.Token);
```

Comment:
@ManickaP I think it'd be better to split the code instead of passing bool parameter as it will avoid generating async infra code for sync methods and there will be no dependency to sync code from async-only workloads

Reply:
Why is this method triggering the warning? This is private, not a public API. I also do not see anything from `HttpClient` in the `unsupported browser warnings.txt` mentioned in the description. So I'm wondering why is this even touched?

I'm asking because the whole implementation of sync is based on this. Down the call chain you'll find many more private methods with `bool async` parameter. So fixing it here might not solve anything and just move the problem one layer down.

Reply:
The warning is triggered by the tool which checks if all code which can possibly throw PNSE is correctly annotated/propagated. The warning fires here correctly because this method can possibly throw PNSE.

We can suppress it like it's done in this PR but we could avoid that if we could propagate the attribute up. This case is interesting because it mixes async and sync code which I think is not ideal because it penalizes the sync version implementation with underlying async infrastructure which could be worth addressing on its own.

Reply:
So you'd split this into 2 private helpers, with the same body except for two lines of code? I'm wondering how would it look like, how much code duplication would be introduced.

We can try that and see how it looks like and decide whether it's worth it.

Reply:
Yep, the helpers would actually be quite different in IL because one of them would be of `async ValueTask<HttpResponseMessage> SendAsyncCore` and second one `HttpResponseMessage SendSyncCore`.

Reply:
> I think it'd be better to split the code instead of passing bool parameter as it will avoid generating async infra code for sync methods and there will be no dependency to sync code from async-only workloads

I disagree.  Splitting the code all the way down will result in _massive_ code duplication, making the code harder to maintain, leading to even larger binary size, and making it much more likely we'll diverge in implementation between the sync and async.  The sync code is there to be a functional alternative when you really can't use async for some reason, and I'm not concerned about the extra async machinery involved on the sync code path.

Reply:
> Splitting the code all the way down will result in massive code duplication

It does not look like massive duplication to me if instead of 50 lines you have 75+ including comments (https://gist.github.com/marek-safar/4700253934660069fb0dbe1cbaf4fb00)

Reply:
You're talking about just the HttpClient layer then?  We use this same bool approach throughput SocketsHttpHandler to have a single implementation of HTTP/1.1 that works for both sync and async.

Reply:
The question is: will the split at the `HttpClient` level solve the problem? Won't it just bubble up at another place? And if it solves it, is it worth it (code duplication of one method vs. warning disable)?

Reply:
Yes, we are talking about this method only (that's why the comment is here) because it's common for all http-handler implementations. In browser-wasm case, it manifests this way but for other implementations it might have size impacts because even if you always use async versions the high-level code pulls sync dependencies as well. 

I think SocketHttpHandler implementation is not relevant here as it's implementation default of the handler and it's just one of the handlers.

Reply:
Are you guys OK with using explicit if condition so that we can assert the non-browser side as @jeffhandley suggested https://github.com/dotnet/runtime/pull/43363#discussion_r504452273

Reply:
@ManickaP Should we just keep this suppression in here for now and file an issue to split this method into Sync/Async methods?

Reply:
I'm OK with what Buyaa suggests, i.e.: explicit if condition with an assert. The suppression is OK as well.
But I'd rather avoid duplicating the calling method as Marek suggested.